File: solutions/clojure/darts/1/src/darts.clj
(ns darts)

(defn score
  "Calculates the score of a dart throw."
  [x y]
  ;; function body
  (let [dist (+ (* x x) (* y y))]
    (cond
      (<= dist 1) 10
      (<= dist 25) 5
      (<= dist 100) 1
      :else 0)
    )
  )

File: solutions/cpp/darts/1/darts.cpp
#include "darts.h"

namespace darts {

  int score(double x, double y) {
    double radsquared = (x * x) + (y * y);
    if (radsquared > 100) {
      return 0;
    } else if (radsquared > 25) {
      return 1;
    } else if (radsquared > 1) {
      return 5;
    } else {
      return 10;
    }
  }

}  // namespace darts
